ksql-engine/src/main/java/io/confluent/ksql/function/UdfLoader.java

```diff
@@ -320,6 +327,92 @@ private static Object instantiateUdfClass(final Method method,
     }
   }
 
+  private static Object instantiateUdfClass(final Class udfClass,
+                                            final UdfDescription annotation) {
+    try {
+      return udfClass.newInstance();
+    } catch (final Exception e) {
+      throw new KsqlException("Failed to create instance for UDF="
+              + annotation.name(), e);
+    }
+  }
+
+  private Function<List<Schema>,Schema> handleUdfReturnSchema(
+          final Class theClass,
+          final Method method,
+          final Udf udfAnnotation,
+          final UdfDescription descAnnotation) {
+
+    final String schemaProviderName = method.getAnnotation(Udf.class).schemaProvider();
+
+    if (!schemaProviderName.equals("")) {
+      return handleUdfSchemaProviderAnnotation(schemaProviderName, theClass, method,
+              descAnnotation);
+    }
+
+    final Schema returnSchema = getReturnType(method, udfAnnotation);
+    if (returnSchema.name() != null
+        && returnSchema.name().equals("org.apache.kafka.connect.data.Decimal")
```

Comment:
nit: use `DecimalUtil#isDecimal` - in general you don't want to be using hard coded string constants so that if it ever changes we only need to change it in one place